client.js

```diff
@@ -1,182 +1,154 @@
-// Generated by CoffeeScript 2.5.1
-(function() {
-  var SugarClient;
-
-  SugarClient = (function() {
-    class SugarClient {
-      constructor(userId, authToken) {
-        this.primusUrl = this.primusUrl.bind(this);
-        this.connect = this.connect.bind(this);
-        this.disconnect = this.disconnect.bind(this);
-        this.receiveData = this.receiveData.bind(this);
-        this.subscribeTo = this.subscribeTo.bind(this);
-        this.unsubscribeFrom = this.unsubscribeFrom.bind(this);
-        this.on = this.on.bind(this);
-        this.off = this.off.bind(this);
-        this.emit = this.emit.bind(this);
-        this.__subscribeToChannels = this.__subscribeToChannels.bind(this);
-        this.__subscribeTo = this.__subscribeTo.bind(this);
-        this.createEvent = this.createEvent.bind(this);
-        this.userId = userId;
-        this.authToken = authToken;
-        this.events = {};
-        this.subscriptions = {};
-        this.initializePrimus();
-      }
-
-      initializePrimus() {
-        if (SugarClient.Primus == null) {
-          throw 'SugarClient.Primus is not defined';
-        }
-        if (SugarClient.host == null) {
-          throw 'SugarClient.host is not defined';
-        }
-        this.primus = SugarClient.Primus.connect(SugarClient.host, {
-          websockets: true,
-          network: true,
-          manual: true,
-          ping: 10000
-        });
-        this.primus.on('outgoing::url', this.primusUrl);
-        return this.primus.on('data', this.receiveData);
-      }
+/*
+ * decaffeinate suggestions:
+ * DS101: Remove unnecessary use of Array.from (note: done by srallen)
+ * DS102: Remove unnecessary code created because of implicit returns
+ * DS103: Rewrite code to no longer use __guard__, or convert again using --optional-chaining (note: done by srallen)
+ * DS205: Consider reworking code to avoid use of IIFEs
+ * DS206: Consider reworking classes to avoid initClass
+ * DS207: Consider shorter variations of null checks
+ * Full docs: https://github.com/decaffeinate/decaffeinate/blob/master/docs/suggestions.md
+ */
+class SugarClient {
+  static initClass() {
+    this.host = null;
+    this.Primus = null;
```

Comment:
Running `npm link` and testing with PFE hit an issue `Uncaught TypeError: Cannot set property 'Primus' of undefined` here. I think related to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes#binding_this_with_prototype_and_static_methods, I'm making some progress troubleshooting, but want to post update asap.

Reply:
@mcbouslog thanks for checking. I'll continue this into the week. It's more important that survey task is finished than this gets decaffeinated.